### PR TITLE
feature/99 - checkPopulate fn for results node populates

### PR DESCRIPTION
**Pull request by doug-patterson**
It's possible to create a hard-to-debug long-running query by trying to populate a field you didn't include with a previous populate or the base includes. Code here includes a fn to check for this situation and throw an error

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 248e100de227bd171497240297606ee9d9aee652


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ad5b33f2ad087e5301bddcd1de55443639d58edd
</p>

**Comment by doug-patterson** [comment]
Still WIP as I realize now I need to make it use the schema fields if node.include isn't there

**Comment by doug-patterson** [comment]
@daedalus28 ready for review

File: test/example-types/results.js [hunk 1]
```diff
@@ -346,4 +347,66 @@ describe('results', () => {
...
+            include: ['_id', 'firstName', 'lastName', 'organization'],
+          },
+          _createdByOrganization: {
+            localField: 'createdBy.organization',
+          },
+        },
+      }
+      expect(() => checkPopulate(node)).not.to.throw()
+    })
+    it('should throw for omitted node.include when schema does not the lookup either', () => {
```
Comment (ccalbaugh) [inline test/example-types/results.js:381]: I'm actually not sure what this is saying, can you clarify? :)

**Review by ccalbaugh — APPROVED** [review]
Looks good as long as that test description gets updated 👍

**Comment by doug-patterson** [comment]
@dshishkov @ccalbaugh @daedalus28 @Akrion ready for review

**Pull request merged**